processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -88,4 +88,23 @@ else if(o instanceof Number)
     }
     throw new IAE("Cannot convert [%s] to QueryGranularity", o.getClass());
   }
+
+  // best effort comparing
+  public Integer compare(QueryGranularity other) {
```

Comment:
With a given interval and granularity, we can make a set of truncated timestamps. If this time-set is a subset of other time-set made from some granularity with the same interval, we might say that the former granularity is bigger than the later and vise-versa. 

I've thought at first that this could be used for some optimizations but it seemed hard to find meaningful situation for using it. Any thoughts?